src/notation/view/noteinputbarmodel.cpp

```diff
@@ -186,6 +189,20 @@ void NoteInputBarModel::onNotationChanged()
         undoStack()->stackChanged().onNotify(this, [this]() {
             updateState();
         });
+
+        async::NotifyList<const Part*> partList = masterNotationParts()->partList();
```

Comment:
It is better to add somewhere (for example, to NotationParts) one additional notification. It will notify us when the list becomes empty or not. So this code (and other similar blocks) can be rewritten as follows:

```
masterNotationParts()->hasPartsChanged(this, [this]() {
    emit isInputAllowedChanged();
});
```